tests/NetEscapades.EnumGenerators.Tests/EnumGeneratorTests.cs

```diff
@@ -583,4 +584,179 @@ public enum MyEnum
         Assert.Empty(diagnostics);
         return Verifier.Verify(output, Settings());
     }
+
+    [Fact]
+    public void GeneratesWarningForEnumInGenericType()
+    {
+        const string input =
+            """
+            using NetEscapades.EnumGenerators;
+
+            namespace MyTestNameSpace
+            {
+                public class GenericClass<T>
+                {
+                    [EnumExtensions]
+                    public enum MyEnum
+                    {
+                        First,
+                        Second,
+                    }
+                }
+            }
+            """;
+
+        // Create a simplified compilation and run the generator
+        var syntaxTree = CSharpSyntaxTree.ParseText(input);
```

Comment:
Don't create a simplified compliation, use the same helpers that are used elsewhere in the files